pkg/wamr/Makefile

```diff
@@ -44,7 +44,11 @@ ifeq ($(findstring aarch,$(OS_ARCH)),aarch)
   WAMR_BUILD_TARGET = ARM
 endif
 else ifeq ($(findstring arm,$(CPU_ARCH)),arm)
-  WAMR_BUILD_TARGET = THUMB
+  ifeq ($(findstring cortexm_fpu,$(FEATURES_PROVIDED)),cortexm_fpu)
```

Comment:
Nit picking: Would it be possible to disable hardware FPU support if that is provided, and is this something that we need to be prepared for?

I think just using `FEATURES_USED` instead of `FEATURES_PROVIDED` would work. But there is a CI test that will fail if anyone uses the `FEAUTRES_USED` variable, which would prevent this from getting merged. I wonder if that CI check has outlived its usefulness anyway.

In either case, I think merging this is an improvement and this concern should not block this. But maybe we could add a follow up, if there is any merit to my concern.

Reply:
Regarding the `FEATURES_USED` vs `FEATURES_PROVIDED`, since from my understanding VFP is an extension and therefore provides the same capabilities as the normal architecture I think that as long it doesn't create performance issues it should be used if the feature is provided.